includes/Plugin.php

```diff
@@ -47,9 +47,43 @@ public static function instance(): self {
 	 * Setup the plugin.
 	 */
 	private function setup(): void {
+		// Bail if dependencies are not met.
+		if ( ! $this->has_dependencies() ) {
+			return;
+		}
+
 		McpAdapter::instance();
```

Comment:
By only hooking into WordPress if dependencies are met, we don't need the registry itself to handle state/availability checks